dali/operators/geometry/affine_transforms/crop_transform.cc

```diff
@@ -0,0 +1,148 @@
+be mapped to ``to_start`` and ``from_end`` will be mapped to ``to_end``.
+
+If another transform matrix is passed as an input, the operator applies the transformation to the matrix provided.
+
+.. note::
+    The output of this operator can be fed directly to the ``MT`` argument of ``CoordTransform`` operator.
+)code")
+  .AddArg(
+    "from_start",
+    R"code(The lower bound of the original coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "from_end",
+    R"code(The upper bound of the original coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "to_start",
+    R"code(The lower bound of the destination coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddArg(
+    "to_end",
+    R"code(The upper bound of the destination coordinate space.)code",
+    DALI_FLOAT_VEC, true)
+  .AddOptionalArg(
+    "absolute",
+    R"code(If set to true, start and end coordinates will be swapped if start > end.)code",
+    false, true)
+  .NumInput(0, 1)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+/**
+ * @brief Scale transformation.
+ */
+class CropTransformCPU
+    : public TransformBaseOp<CPUBackend, CropTransformCPU> {
+ public:
+  using SupportedDims = dims<1, 2, 3, 4, 5, 6>;
+
+  explicit CropTransformCPU(const OpSpec &spec) :
+      TransformBaseOp<CPUBackend, CropTransformCPU>(spec),
+      from_start_("from_start", spec),
+      from_end_("from_end", spec),
+      to_start_("to_start", spec),
+      to_end_("to_end", spec) {
+  }
+
+  template <typename T, int mat_dim>
+  void DefineTransforms(span<affine_mat_t<T, mat_dim>> matrices) {
+    constexpr int ndim = mat_dim - 1;
+    assert(matrices.size() == static_cast<int>(from_start_.size()));
+    assert(matrices.size() == static_cast<int>(from_end_.size()));
+    assert(matrices.size() == static_cast<int>(to_start_.size()));
+    assert(matrices.size() == static_cast<int>(to_end_.size()));
+    for (int i = 0; i < matrices.size(); i++) {
+      auto &mat = matrices[i];
+      const vec<ndim> &from_start = *reinterpret_cast<vec<ndim>*>(from_start_[i].data());
+      const vec<ndim> &from_end = *reinterpret_cast<vec<ndim>*>(from_end_[i].data());
+      const vec<ndim> &to_start = *reinterpret_cast<vec<ndim>*>(to_start_[i].data());
+      const vec<ndim> &to_end = *reinterpret_cast<vec<ndim>*>(to_end_[i].data());
+      mat = affine_mat_t<T, mat_dim>::identity();
+      vec<ndim> from_extent = from_end - from_start;
+      vec<ndim> to_extent = to_end - to_start;
+      vec<ndim> offset = to_start - from_start;
+
+      if (absolute_) {
+        for (int d = 0; d < ndim; d++) {
+          from_extent[d] = std::abs(from_extent[d]);
+          to_extent[d] = std::abs(to_extent[d]);
+        }
+      }
+
+      for (int d = 0; d < ndim; d++) {
+        constexpr float kEps = std::numeric_limits<float>::epsilon();
+        DALI_ENFORCE(std::abs(from_extent[d]) > kEps,
+          make_string("from_end[d] should be different from from_start[d] for all "
```

Comment:
What if from_extent <= kEps, but from_start and from_end differs. The user would be confused.

Reply:
if the difference is smaller than epsilon, we consider it as equal. Do you see any issue with that?

Reply:
I mean that the error message would be confusing, the user would see that they differ but for DALI it does not.
Maybe write that it should differ by at least `XXX` and print this kEps. (I don't know if we cannot hit an edge case where the results of subtraction would be rounded to be smaller than kEps but for the user it will be bigger).

Reply:
How about now?